### ISSUE TITLE
gobgp's neighbor config needs local-address 

### ISSUE THREAD

**Issue by tamihiro** [cloudnativelabs/kube-router#778]
My lab setup has nodes with multiple L3 interfaces, and each node advertises from a loopback interface address to others.

- node3:
```
$ ip a l | grep -E '^[1-9]|inet\b'
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    inet 127.0.0.1/8 scope host lo
    inet 192.168.0.13/32 scope global lo
2: ens3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    inet 192.168.10.251/29 brd 192.168.10.255 scope global ens3
3: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    inet 192.168.30.251/29 brd 192.168.30.255 scope global ens4
4: kube-bridge: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.97.0.1/24 scope global kube-bridge
5: veth109580db@if3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master kube-bridge state UP group default 
6: kube-dummy-if: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default 
    inet 10.96.0.10/32 brd 10.96.0.10 scope link kube-dummy-if
    inet 10.96.0.1/32 brd 10.96.0.1 scope link kube-dummy-if
7: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
8: tun-192168014@lo: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 65496 qdisc noqueue state UNKNOWN group default qlen 1000
```

- node4:
```
$ ip a l | grep -E '^[1-9]|inet\b'
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    inet 127.0.0.1/8 scope host lo
    inet 192.168.0.14/32 scope global lo
2: ens3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    inet 192.168.20.251/29 brd 192.168.20.255 scope global ens3
3: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    inet 192.168.40.251/29 brd 192.168.40.255 scope global ens4
4: kube-bridge: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.97.1.1/24 scope global kube-bridge
5: vethc4a0377a@if3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master kube-bridge state UP group default 
6: kube-dummy-if: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default 
    inet 10.96.0.10/32 brd 10.96.0.10 scope link kube-dummy-if
    inet 10.96.0.1/32 brd 10.96.0.1 scope link kube-dummy-if
7: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
8: tun-192168013@lo: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 65496 qdisc noqueue state UNKNOWN group default qlen 1000
```

- api servers
```
# kubectl get pod -n kube-system kube-apiserver-node3 -o custom-columns=CONTAINER:.spec.containers[0].command[1]
CONTAINER
--advertise-address=192.168.0.13
# kubectl get pod -n kube-system kube-apiserver-node4 -o custom-columns=CONTAINER:.spec.containers[0].command[1]
CONTAINER
--advertise-address=192.168.0.14
```

I started kube-router daemonset, but they didn't establish peering. Log messages below explain why they fail.

- from node3
```
time="2019-09-03T15:27:51Z" level=info msg="Can't find configuration for a new passive connection from:192.168.20.251" Topic=Peer
```

- from node4
```
time="2019-09-03T15:27:53Z" level=info msg="Can't find configuration for a new passive connection from:192.168.30.251" Topic=Peer
```

So node3 has been configured such that the address of its neighbor (node4) is `192.168.0.14`, but node4 sends packets to node3 from `192.18.20.251`. And node3 sends packets to node4 likewise.

For kube-router to be able to work in this scenario, we need to add `local-address` to transport config of each neighbor.
I've submitted #777 as a fix. With my PR `local-address` is configured for every neighbor even when it isn't necessary, but I believe it doesn't hurt anything.

**Comment by aauren** [comment]
@murali-reddy these two PRs basically make it so that kube-router specifies the local address to use when peering and doesn't specify the device when the interface is localhost.

While I think that the use-case here is kinda niche, I can't think of any reason why it would be bad to be a little more specific.

Can you give your thoughts? If I don't hear anything, I'll probably ask for a rebase and merge after a week (or more realistically, whenever I next go through issues and PRs 😛 )

**Comment by aauren** [comment]
Closed via #777 and #779
